### ISSUE TITLE
Missing Service Create command(s) in 10-deploy-model.ipynb?

### ISSUE THREAD

**Issue by jaxidian** [Azure/MachineLearningSamples-EnergyDemandTimeSeriesForecasting#7]
Is there a missing command(s) in 10-deploy-model.ipynb? I see the sequence is this:

1. az ml service show realtime ...
2. az ml service run realtime ...
3. az ml service delete realtime ...

However, I would expect that sequence to begin with a `az ml service create realtime ...` line and it seems that would exist before the text: "Check web service is running."

Am I missing something? I have followed the steps on the [Model Management Setup guide](https://docs.microsoft.com/en-us/azure/machine-learning/preview/model-management-configuration) up until the final command where it runs `az ml service create realtime ...` as I had no model to actually deploy when I was going through the pre-reqs and it was my understanding that this `10-deploy-model.ipynb` notebook would do the actual deployment.

**Comment by jaxidian** [comment]
As best as I can tell, it was removed with this commit:

https://github.com/Azure/MachineLearningSamples-EnergyDemandTimeSeriesForecasting/commit/e0f91c2cb8ab649b2ce5b17179895265d1ad6797

**Comment by jaxidian** [comment]
I have tried both the command that was removed in that commit as well as a slightly modified one and neither are working for me.

Here's the modified attempt:

```
!az ml service create realtime --model-file model_deploy.pkl -f score.py -n demandforecast -s service_schema.json -r python
```

The output:

```
 model_deploy.pkl
Successfully registered model
Id: 8530beed920e4e2a8079826ab20814f2
More information: 'az ml model show -m 8530beed920e4e2a8079826ab20814f2'
Creating new driver at C:\Users\SHANEM~1\AppData\Local\Temp\tmpl15i3lta.py
 score.py
 service_schema.json
Successfully created manifest
Id: 93d132a2-1397-4616-84b2-28308641c42e
More information: 'az ml manifest show -i 93d132a2-1397-4616-84b2-28308641c42e'
Creating image.........Done.
Image ID: 89ca50bb-2ef9-497b-b320-c5d8c7df3ce9
More details: 'az ml image show -i 89ca50bb-2ef9-497b-b320-c5d8c7df3ce9'
Usage information: 'az ml image usage -i 89ca50bb-2ef9-497b-b320-c5d8c7df3ce9'
[Local mode] Running docker container.
[Local mode] Pulling the image from mlcrpacr87d66978e808.azurecr.io/demandforecast:3. This may take a few minutes, depending on your connection speed...
[Local mode] Pulling

ERROR: {'Error': MlCliError({'Response Content': APIError(HTTPError('500 Server Error: Internal Server Error for url: http+docker://localnpipe/v1.30/images/create?tag=3&fromImage=mlcrpacr87d66978e808.azurecr.io%2Fdemandforecast',),), 'Error': 'Unable to pull image mlcrpacr87d66978e808.azurecr.io/demandforecast:3'},), 'Azure-cli-ml Version': '0.1.0a27.post3'}
```

**Comment by angusrtaylor** [comment]
Thanks for spotting this. The command to create the web service was missing. I'll add it to the notebook. The command is:

!az ml service create realtime --model-file model_deploy.pkl -f score.py -n demandforecast -s service_schema.json -r python

It should work provided you have Docker running; have run 2-linear-regression.ipynb to train the model; and are running 10-deploy-model.ipynb with the "PROJECT_NAME local" kernel. Please let me know if it is still not working.

**Comment by myamama** [comment]
i get the following error:
ERROR: Unable to start container: 500 Server Error: Internal Server Error ("Get https://XXXXX.azurecr.io/v2/demandforecast/manifests/1: no basic auth credentials")

**Comment by angusrtaylor** [comment]
@myamama please can you try restarting Docker on your machine? I had a similar issue today and restarting Docker resolved it. Thanks.

**Comment by jaxidian** [comment]
@angusrtaylor I tried with that command and was able to reproduce the error. I then went through a reboot and started the entire tutorial from scratch. This second time, when I reached this step, it churned for minutes and then Windows Defender finally gave me a popup to allow network access to Docker. I granted this permission and it finally succeeded. I failed to screenshot it for details (sorry) but I think that was the cause of my 500 Server Error, although I'm not quite sure why it resulted in a 500.

FYI, this was a pretty clean install of (latest stable) Windows 10. I installed Windows 10, joined an AAD, installed Visual Studio 2017, VSCode, Office, and all Windows Updates & Dell drivers (this is a Dell XPS 13 9350 I'm running all of this on). Once all of that was fully updated, I then went through your tutorial and started downloading/installing/configuring the pre-reqs. All was well up until that point of the `10-deploy-model.ipynb` notebook (and I definitely had `PROJECT_NAME local` selected as the kernel. Also, Docker was definitely running successfully. All of this is to say that, I think if somebody has a clean Docker install on a clean Windows 10 install, this firewall hole is an important thing to consider for the tutorial for your readers.
